Lib/test/test_shelve.py

```diff
@@ -222,22 +224,25 @@ def deserializer(data):
                         s["array_data"], array_data.tobytes().decode()
                     )
 
-    def test_custom_incomplete_serializer_and_deserializer(self):
-        dbm_sqlite3 = import_helper.import_module("dbm.sqlite3")
+    def test_custom_incomplete_serializer(self):
         os.mkdir(self.dirname)
         self.addCleanup(os_helper.rmtree, self.dirname)
 
-        with self.assertRaises(dbm_sqlite3.error):
-            def serializer(obj, protocol=None):
-                pass
+        def serializer(obj, protocol=None):
+            pass
 
-            def deserializer(data):
-                return data.decode("utf-8")
+        def deserializer(data):
+            return data.decode("utf-8")
 
+        with self.assertRaises((TypeError, dbm.error)):
```

Comment:
```suggestion
        # Since the serializer returns None, dbm.error is raised
        # by dbm.sqlite3 and TypeError is raised by other backends.
        with self.assertRaises((TypeError, dbm.error)):
```

and copy this comment below.